src/client.rs

```diff
@@ -78,14 +78,15 @@ impl ClientConfig {
 
 // ClientState represents the internal state of the client at any given point in
 // time.
+#[derive(Debug)]
 pub struct ClientState {
     pub current_nick: String,
     pub config: Arc<ClientConfig>,
 }
 
 // Client represents the running bot.
 pub struct Client {
-    plugins: Vec<Box<dyn Plugin>>,
+    plugin_senders: Mutex<Vec<mpsc::Sender<Arc<Context>>>>,
```

Comment:
Sorry, missing in the code here. Why does this need a mutex? I get that each task is running separately, but why does the queue of senders need the mutex?

Reply:
It needs a mutex because you need to be able to access the senders mutably and the Client is behind an arc